src/hotspot/os/windows/os_windows.hpp

```diff
@@ -188,18 +188,21 @@ class PlatformEvent : public CHeapObj<mtSynchronizer> {
 
 
 
-class PlatformParker : public CHeapObj<mtSynchronizer> {
-  protected:
-    HANDLE _ParkEvent ;
+class PlatformParker {
+  NONCOPYABLE(PlatformParker);
 
-  public:
-    ~PlatformParker () { guarantee (0, "invariant") ; }
-    PlatformParker  () {
-      _ParkEvent = CreateEvent (NULL, true, false, NULL) ;
-      guarantee (_ParkEvent != NULL, "invariant") ;
-    }
+ protected:
+  HANDLE _ParkEvent ;
 
-} ;
+ public:
+  PlatformParker() {
+    _ParkEvent = CreateEvent (NULL, true, false, NULL) ;
+    guarantee(_ParkEvent != NULL, "invariant") ;
+  }
+  ~PlatformParker() {
+    CloseHandle(_ParkEvent);
+  }
+};
```

Comment:
Should the implementation of the constructor and destructor be in the .cpp file so that thread.hpp doesn't need to include whatever windows os specific file needed to import CreateEvent and CloseHandle?

Reply:
I see there's a lot more work to minimize the include dependencies here.